0.5.0

peppelinux opened the pull request:
Proofreading and small editorials for the release number 0.5.0

peppelinux committed:
feat: remove duplicated actions in the CI

peppelinux committed:
Merge branch 'versione-corrente' of https://github.com/italia/eidas-it-wallet-docs into versione-corrente

peppelinux committed:
chore: introduction, working

peppelinux committed:
chore: ssi-introduction

peppelinux committed:
chore: trust model, wording

peppelinux committed:
chore: pid model, wording

peppelinux committed:
chore: issuance, wording

peppelinux committed:
chore: wallet solution, wording

peppelinux committed:
chore: wia, wording

peppelinux committed:
chore: RP, wording and acknowledgments

peppelinux committed:
fix: html linting

peppelinux committed:
chore: RP - tationale behind DPOP

peppelinux committed:
fix: typo

peppelinux committed:
fix: introduction

peppelinux committed:
fix: introduction

peppelinux committed:
fix: introduction

peppelinux committed:
fix: introduction

peppelinux committed:
fix: introduction

peppelinux committed:
fix: title

Sh-Amir commented on docs/en/defined-terms.rst:
```diff
@@ -51,6 +51,8 @@ Below are the description of acronyms and definitions which are useful for furth
    * - Level of Assurance
      - The degree of confidence in the vetting process used to establish the identity of the User and the degree of confidence that the User who presents the credential is the same User to whom the credential was issued.
+   * - Holder Key Binding
+     - Ability of the Holder to prove legitimate possession of the private part, related to the public part attested by a Trusted Third Party.
```
It is a very minor comment. 
However, I recommend for clarity substituting the ``private part`` with ``private key`` and ``public part`` with ``public key``.

Sh-Amir replied:
Would it be valuable to add the definition of the `Trusted Third Party` as the following deriving from [FIPS 186-5](https://doi.org/10.6028/NIST.FIPS.186-5), or [NIST SP 800-89](https://doi.org/10.6028/NIST.SP.800-89):

- `An entity other than the key pair owner (Holder) and Verifier that is trusted by the Holder or the Verifier or both. Sometimes shortened to “Trusted Party.`
- ` An entity other than the Holder and Verifier that is trusted by the Holder, the Verifier or both to provide certain services.`

peppelinux replied:
Ok, see my suggestion

Sh-Amir commented on docs/en/index.rst:
```diff
@@ -8,18 +8,19 @@ Introduction
 The purpose of the following technical rules is to define the technical architecture and reference framework to be used as a guideline by all the parties involved in the development of the IT Wallet project.
 
-In this documentation you can find the technical specification for implementing the components that compose the Wallet ecosystem:
+In this documentation there are the technical details for implementing the components that compose the Wallet ecosystem:
```
[Minor recommendation]:
`This document provides technical guidelines for the secure implementation of Wallet ecosystem components that include:'

peppelinux replied:
Done

Sh-Amir commented on docs/en/pid-eaa-data-model.rst:
```diff
@@ -7,8 +7,7 @@ PID/(Q)EAA Data Model
 The Person Identification Data (PID) is issued by the PID Provider following national laws and allows a natural person to be authenitcated and identified. 
-
-The User attributes carried in the Italian PID are:
+The User attributes carried within the Italian PID are the ones listed below:
```
In the new version of the ARF 1.2, the mandatory set of user attributes for the PID has been changed, do we need to be compliant with that? they considered `age_over_18` as a mandatory attribute in the Italian PID?

peppelinux replied:
good catch @asharif1990 

moved to a separate issue

Sh-Amir commented on docs/en/pid-eaa-issuance.rst:
I am working on the security checks that must be performed by the Wallet Instance and PID/(Q)EAA issuer during the issuance phase. I need some more days to complete it. Would you like to go for the release without them or as soon as they are complete, I will do a pull request for the current release?

peppelinux replied:
yes, we'll have it in 0.6.0

I'd like to have this in 0.5.0:
https://github.com/italia/eudi-wallet-it-docs/issues/101

peppelinux committed:
fix: title

grausof reviewed (approved):
LGTM

peppelinux commented on docs/en/index.rst:
```diff
@@ -1,33 +1,33 @@
 The purpose of the following technical rules is to define the technical architecture and reference framework to be used as a guideline by all the parties involved in the development of the IT Wallet project.
 
-In this documentation you can find the technical specification for implementing the components that compose the Wallet ecosystem:
+This documentation defines the national implementation profile of EUDI Wallet, containing the technical details about components of the Wallet ecosystem, as listed below:
```
This document provides technical guidelines for the secure implementation of Wallet ecosystem components that include:

peppelinux replied:
```suggestion
This document provides technical guidelines for the secure implementation of Wallet ecosystem components that include:
```

fmarino-ipzs commented on docs/en/index.rst:
```diff
@@ -1,33 +1,33 @@
 
  - Entities of the ecosystem according to `EIDAS-ARF`_.
  - Infrastructure of trust attesting realiability and eligibility of the participants.
  - PID and EAAs data schemes and attribute sets.
  - PID/EAA in MDL CBOR format.
  - PID/EAA in `SD-JWT`_ format.
```
```suggestion
 - PID/(Q)EAAs data schemes and attribute sets.
 - PID/(Q)EAA in MDL CBOR format.
 - PID/(Q)EAA in `SD-JWT`_ format.
```

fmarino-ipzs commented on docs/en/index.rst:
```diff
@@ -1,33 +1,33 @@
  - Wallet Solution general architecture.
- - Wallet Instance Attestation data model in `JWS`_ format.
+ - Wallet Instance Attestation.
  - Issuance of PID/EAA according to `OpenID4VCI`_.
  - Presentation of PID/EAA according to `OpenID4VP`_.
```
```suggestion
 - Issuance of PID/(Q)EAA according to `OpenID4VCI`_.
 - Presentation of PID/(Q)EAA according to `OpenID4VP`_.
```

fmarino-ipzs commented on docs/en/pid-eaa-data-model.rst:
```diff
@@ -23,7 +22,7 @@ The (Q)EAAs are issued by (Q)EAA Issuers to a Wallet Instance and MUST be provid
 The (Q)EAAs are extended according to the `OpenID Identity Assurance Profile [OIDC.IDA] <https://openid.net/specs/openid-connect-4-identity-assurance-1_0-13.html>`_, that allows the recipients to know the Authentic Sources where the data comes from. 
 
-The PID/(Q)EAA data format and the mechanism through which a digital credential is issued to the Wallet Instance and presented to an RP is described in the following sections. 
+The PID/(Q)EAA data format and the mechanism through which a digital credential is issued to the Wallet Instance and presented to an Relying Party is described in the following sections. 
```
```suggestion
The PID/(Q)EAA data format and the mechanism through which a digital credential is issued to the Wallet Instance and presented to a Relying Party are described in the following sections. 
```

fmarino-ipzs replied:
```suggestion
The PID/(Q)EAA data format and the mechanism through which a digital credential is issued to the Wallet Instance and presented to a Relying Party are described in the following sections. 
```

fmarino-ipzs commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -63,18 +63,17 @@ The :numref:`fig_High-Level-Flow-ITWallet-QEAA-Issuance` shows a general archite
 Below the description of the most relevant operations involved in the (Q)EAA issuance:
 
-    1. **Obtaining the trusted (Q)EAA Issuer**: the Wallet Instance discovers the trusted (Q)EAA Issuer at the Subordinate Listing Endpoint of the Trust Anchor, and then inspects the metadata looking for the credential capabilities. 
+    1. **Obtaining the trusted (Q)EAA Issuer**: the Wallet Instance discovers the trusted (Q)EAA Issuer ufing the Federaiton API (e.g.: using the Subordinate Listing Endpoint of the Trust Anchor and its Intermediates), then inspects the metadata looking for the credential capabilities. 
```
```suggestion
    1. **Obtaining the trusted (Q)EAA Issuer**: the Wallet Instance discovers the trusted (Q)EAA Issuer using the Federation API (e.g.: using the Subordinate Listing Endpoint of the Trust Anchor and its Intermediates), then inspects the metadata looking for the credential capabilities. 
```

fmarino-ipzs commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -240,7 +238,7 @@ The JWS payload of the request object is represented below:
 .. note::
 
-    **PID Credential Schema and Status registration:** The PID/(Q)EAA Issuer MUST register all the issued credentials for their later revocation, if needed. 
+    **PID Credential Schema and Status registration:** The PID/(Q)EAA Issuer MUST register all the issued credentials for their later revocation. 
```
```suggestion
    **PID/(Q)EAA Credential Schema and Status registration:** The PID/(Q)EAA Issuer MUST register all the issued credentials for their later revocation. 
```

fmarino-ipzs commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -664,17 +661,18 @@ A DPoP-bound Access Token is provided by the PID/(Q)EAA Token endpoint as a resu
 -------------------
 
-The Credential Endpoint issues a Credential as approved by the End-User upon presentation of a valid Access Token representing this approval, as defined in `OPENID4VCI`_.
+The Credential Endpoint issues a credential as approved by the End-User upon presentation of a valid Access Token representing this approval, as defined in `OPENID4VCI`_.
```
@peppelinux , should credential be a defined term?

peppelinux replied:
no, since it is defined in this section with a dedicated section

fmarino-ipzs commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -664,17 +661,18 @@ A DPoP-bound Access Token is provided by the PID/(Q)EAA Token endpoint as a resu
 The Wallet Instance when requests the PID/(Q)EAA to the PID/(Q)EAA Credential endpoint, MUST use the following parameters in the entity-body of the HTTP POST request, using the `application/json` media type.
-The Credential endpoint MUST accept and validate the *DPoP proof* sent in the DPoP field of the Header based on the steps defined in Section 4.3 of [DPoP-draft16]. The *DPoP proof* in addition to the values that are defined in the Token Endpoint section MUST contain the following claim:
+
+The Credential endpoint MUST accept and validate the *DPoP proof* sent in the DPoP field of the HTTP Header based on the steps defined in Section 4.3 of [DPoP-draft16]. The *DPoP proof* in addition to the values that are defined in the Token Endpoint section MUST contain the following claim:
```
```suggestion
The Credential Endpoint MUST accept and validate the *DPoP proof* sent in the DPoP field of the HTTP Header based on the steps defined in Section 4.3 of [DPoP-draft16]. The *DPoP proof* in addition to the values that are defined in the Token Endpoint section MUST contain the following claim:
```

fmarino-ipzs commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -763,13 +760,13 @@ Credential Response to the Wallet Instance MUST be sent using `application/json`
     - `[OIDC4VCI. Draft 13] <https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html>`_.
   * - **credential**
-    - Contains the issued PID/(Q)EAA. It MUST be an SD-JWT JSON Object (see Section :ref:`PID/(Q)EAA Data Model <pid_eaa_data_model.rst>`).
+    - Contains the issued PID/(Q)EAA. When the credential type is SD-JWT, it MUST be an SD-JWT JSON Object (see Section :ref:`PID/(Q)EAA Data Model <pid_eaa_data_model.rst>`).
```
```suggestion
    - Contains the issued PID/(Q)EAA. When the credential type is SD-JWT, it MUST be an `SD-JWT`_ JSON Object (see Section :ref:`PID/(Q)EAA Data Model <pid_eaa_data_model.rst>`).
```

peppelinux commented on docs/en/defined-terms.rst:
```diff
@@ -51,6 +51,8 @@ Below are the description of acronyms and definitions which are useful for furth
    * - Level of Assurance
      - The degree of confidence in the vetting process used to establish the identity of the User and the degree of confidence that the User who presents the credential is the same User to whom the credential was issued.
+   * - Holder Key Binding
+     - Ability of the Holder to prove legitimate possession of the private part, related to the public part attested by a Trusted Third Party.
```
```suggestion
     - Ability of the Holder to prove legitimate possession of the private key, related to the public key attested by a Trusted Third Party.
```

peppelinux replied:
```suggestion
     - Ability of the Holder to prove legitimate possession of the private key, related to the public key attested by a Trusted Third Party.
   * - Trusted Third Party
     - An entity other than the Holder and Verifier that is trusted by the Holder, the Verifier or both to provide certain services ([NIST SP 800-89](https://doi.org/10.6028/NIST.SP.800-89)).
```

peppelinux committed:
added contributors and stadards section

peppelinux committed:
fix: standards section

Sh-Amir committed:
[Editorial and minor changes]

[Editorial and minor changes on the PID/(Q)EAA Issuance. Integrate the Security Checks that MUST be performed during the issuance and define how the keys for holder binding must be used. The major change is the usage of JWK for the Holder Binding]

Sh-Amir committed:
[update references]

correct the references.

Sh-Amir committed:
[add the new MSC]

Sh-Amir committed:
[update links + add gitignore]

Sh-Amir committed:
[minor edit]

Sh-Amir committed:
[check cross ref]

Sh-Amir committed:
[minor changes]

Sh-Amir committed:
[fix minor]

Sh-Amir committed:
[minor fixes]

Sh-Amir committed:
minor

Sh-Amir committed:
[check]

Sh-Amir committed:
[update endpoint description section]

Sh-Amir committed:
[minor edits]

Sh-Amir committed:
[minor]

Sh-Amir committed:
Update pid-eaa-issuance.rst

Sh-Amir committed:
[minor]

Sh-Amir committed:
[minor]

Sh-Amir committed:
[minor]

fmarino-ipzs commented on docs/en/pid-eaa-issuance.rst:
```diff
@@ -240,7 +238,7 @@ The JWS payload of the request object is represented below:
 .. note::
 
-    **PID Credential Schema and Status registration:** The PID/(Q)EAA Issuer MUST register all the issued credentials for their later revocation, if needed. 
+    **PID Credential Schema and Status registration:** The PID/(Q)EAA Issuer MUST register all the issued credentials for their later revocation. 
```
```suggestion
    **PID/(Q)EAA Credential Schema and Status registration:** The PID/(Q)EAA Issuer MUST register all the issued credentials for their later revocation. 
```

fmarino-ipzs replied:
We might remove this and deal with it in a new section on credential lifecycle, @peppelinux  what do you think?

peppelinux replied:
@fmarino-ipzs to be addressed in a separate issue

Sh-Amir committed:
[final changes]

fmarino-ipzs commented on docs/en/trust.rst:
```diff
@@ -605,33 +601,34 @@ The Wallet Instance that verifies the request issued by the Relying Party MUST u
 
-Privacy Considerations
-----------------------
+Privacy Final Remarks
+---------------------
```
```suggestion
Privacy Remarks
---------------
```

Sh-Amir committed:
update aud claim

Sh-Amir committed:
[minor]

Sh-Amir committed:
[final changes]

Commit:
Apply suggestions from code review

Co-authored-by: fmarino-ipzs <77629526+fmarino-ipzs@users.noreply.github.com>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

Sh-Amir commented on docs/common/standards.rst:
```diff
@@ -53,7 +53,7 @@ Technical References
-      - D. Fett, B. Campbell, J. Bradley, T. Lodderstedt, M. Jones, D. Waite, "OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer (DPoP)"
-    * - `OPENID4VC-HAIP`
+    * - `DPOP`_
+      - D\. Fett, B. Campbell, J. Bradley, T. Lodderstedt, M. Jones, D. Waite, "OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer (DPoP)"
```
DPoP becomes an RFC as well, shall we update it and add the tag for RFC9449?

fmarino-ipzs replied:
Yes, please.

peppelinux replied:
Done e0248809479b4e122f4653e1f934b8f5c57ce208

Sh-Amir commented on docs/en/relying-party-solution.rst:
```diff
@@ -21,13 +18,19 @@ In this section the following flows are described:
+.. note::
+    The provisioning of the Wallet Instance Attestation from the Wallet Instace to the Relying Party is 
+    under discussion within the international standardization working groups. At the current stage of the draft of this implementation profile,
+    a mechanism based on `OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer (DPoP) <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop>`_
```
Update the DPoP draft into RFC9449

peppelinux replied:
it redirects automatically to the last version (RFC)

Sh-Amir commented on docs/en/relying-party-solution.rst:
```diff
@@ -53,33 +56,33 @@ The details of each step shown in the previous picture are described in the tabl
 -----------------------------
 
-The Relying Party creates a signed request object, then gives it to the Wallet Instance through a HTTP URL (request URI) that points to the resource where the signed request object MUST be available for download. The URL parameters contained in the Relying Party response, containing the request URI, are described in the Table below.
+The Relying Party MUST create a request object in the format of signed JWT, then provide it to the Wallet Instance through an HTTP URL (request URI). The HTTP URL points to the web resource where the signed request object is available for download. The URL parameters contained in the Relying Party response, containing the request URI, are described in the Table below.
```
request object --> Request Object

peppelinux replied:
Done a601d8a9132305884f2c9e208dc0259a81721e4a

Sh-Amir commented on docs/en/relying-party-solution.rst:
```diff
@@ -268,7 +272,7 @@ Therein a non-normative example of the DPoP decoded content:
 --------------------
 
-The Relying Party issues a signed request object, where a non-normative example in the form of decoded header and payload is shown below:
+The Relying Party issues the signed request object, where a non-normative example in the form of decoded header and payload is shown below:
```
request object --> Request Object

peppelinux replied:
Done a601d8a9132305884f2c9e208dc0259a81721e4a

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: fmarino-ipzs <77629526+fmarino-ipzs@users.noreply.github.com>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: fmarino-ipzs <77629526+fmarino-ipzs@users.noreply.github.com>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: fmarino-ipzs <77629526+fmarino-ipzs@users.noreply.github.com>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: fmarino-ipzs <77629526+fmarino-ipzs@users.noreply.github.com>

Sh-Amir committed:
Update docs/en/pid-eaa-issuance.rst

Co-authored-by: fmarino-ipzs <77629526+fmarino-ipzs@users.noreply.github.com>

Commit:
Merge branch '0.5.0' into fix-pid-issuance-doc

Commit:
Merge pull request #126 from asharif1990/fix-pid-issuance-doc

Editorial PID issuance revisions

peppelinux committed:
fix: DPOP is RFC

peppelinux committed:
fix: Request Object
